### PR TITLE
Closes Issue20

### PR DESCRIPTION
**Pull request by MiTo0o**
- [x] #20 
- [x] Remove lines 54 / 55 in __init__.py where error checking/throwing for this error occurs, and replace with padding logic as above.
- [x] Add unit tests for padding logic to ensure all edge cases are accounted for

Concerns?:

- added .venv since a lot of people also use that to name their virtual envs, don't know if that's ok with you
- added warnings library (comes with python) to warn users instead of raising errors in some situations
- in the board.raw() function maybe have `len(charList)` in it's own varible? since it's used so much.

### CODE REVIEW

File: vestaboard/__init__.py [hunk 1]
```diff
@@ -61,6 +62,26 @@ def raw(self, charList):
       for j, char in enumerate(row):
         if not isinstance(char, int):
           raise ValueError(f'Nested lists must contain numbers only - check row {i} char {j} (0 indexed)')
+    if len(charList) > 6:
+      del charList[6:]
```
Comment (SonicRift) [inline vestaboard/__init__.py:66]: Normally I would advise against modifying the input array (side effects and all), but in this case I think it's a brilliant way around the fact that the `.raw()` function doesn't return anything - clever! 🧠

File: tests/test_init.py [hunk 2]
```diff
@@ -68,13 +72,100 @@ def test_valid_raw_input_does_not_fail():
...
+    vb.raw(small_board, pad='center')
+    expected = [
+        [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
+        [0, 0, 0, 0, 0, 0, 5, 0, 0, 6, 0, 0, 0, 0, 0, 0, 0, 0, 2, 0, 0, 0],
+        [1, 2, 3, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
+        [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 4, 0, 0, 1, 0, 0, 0],
+        [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
+        [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0]
+    ]
+    assert small_board == expected
```
Comment (SonicRift) [inline tests/test_init.py:91]: Let's add the blank padding lines here between tests for consistency - some of the lower ones have them, but not all.

File: tests/test_init.py [hunk 3]
```diff
@@ -68,13 +72,100 @@ def test_valid_raw_input_does_not_fail():
...
+        [1, 2, 3, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
+        [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 4, 0, 0, 1, 0, 0, 0],
+        [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
+        [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
+        [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
+    ]
+    assert small_board == expected
+
+def test_no_pad():
+    small_board = return_valid_small_board()
```
Comment (SonicRift) [inline tests/test_init.py:139]: Let's also test that the right warning is being thrown on this and the next. You can use this context for that, it's similar to the assertion that an error will be thrown:
`with pytest.warns(UserWarning):`

**Review by SonicRift — CHANGES_REQUESTED** [review]
This is excellent! 🎉 Nice job on the code and the tests! I've made a few suggestions for your consideration. Planning to add this in, and release a new version!

File: tests/test_init.py [hunk 4]
```diff
@@ -68,13 +72,100 @@ def test_valid_raw_input_does_not_fail():
...
-    i = vestaboard.Installable(apiKey=apiKey, apiSecret=apiSecret, getSubscription=False, saveCredentials=False)
-    vb = vestaboard.Board(i, subscriptionId=subId)
+    i = vestaboard.Installable(apiKey=TEST_API_KEY, apiSecret=TEST_API_SECRET, getSubscription=False, saveCredentials=False)
+    vb = vestaboard.Board(i, subscriptionId=TEST_SUB_ID)
     vb.post('Should not error')
+def test_odd_length_center_pad():
+    small_board = return_valid_small_board()
+    i = vestaboard.Installable(apiKey=TEST_API_KEY, apiSecret=TEST_API_SECRET, getSubscription=False, saveCredentials=False)
+    vb = vestaboard.Board(i, subscriptionId=TEST_SUB_ID)
+    vb.raw(small_board, pad='center')
```
[the remark this answers is not in the document]
Comment (MiTo0o) [reply tests/test_init.py]: didnt do it this way, maybe it's still ok?

**Review by SonicRift — APPROVED** [review]
Nice work on this, it looks great!
